### PR TITLE
Handle SSL errors with retries

### PR DESCRIPTION
**Pull request by sergiyby**
This was already reviewed once at https://codereview.chromium.org/1522953002.

### CODE REVIEW

**Comment by sergiyby** [comment]
This is going to address issues #71, #97, #158.

File: googleapiclient/http.py [hunk 1]
```diff
@@ -546,14 +547,21 @@ def next_chunk(self, num_retries=0):
         }
     http = self._request.http
 
+    resp = None
```
Comment (nathanielmanistaatgoogle) [inline googleapiclient/http.py]: It's not clear to me why this line is necessary? What breaks if it is removed?

File: googleapiclient/http.py [hunk 2]
```diff
@@ -799,16 +815,24 @@ def next_chunk(self, http=None, num_retries=0):
...
-                                     headers=start_headers)
+              'Retry #%d for resumable URI request: %s %s%s'
+              % (retry_num, self.method, self.uri,
+                 ', following status: %d' % resp.status if resp else ''))
+
+        try:
+          resp, content = http.request(self.uri, method=self.method,
+                                       body=self.body,
+                                       headers=start_headers)
+        except ssl.SSLError:
```
Comment (nathanielmanistaatgoogle) [inline googleapiclient/http.py]: This logic appears nearly identically in three places - please factor it out into a module-scope module-private helper function?

**Comment by nathanielmanistaatgoogle** [comment]
Thank you for the contribution!

File: googleapiclient/http.py (hunk 1 above)
Comment (sergiyby) [inline googleapiclient/http.py]: When SSLError happens, resp is not assigned and thus not defined which breaks line 556 which tries to use it in a condition.

File: googleapiclient/http.py (hunk 2 above)
Comment (sergiyby) [inline googleapiclient/http.py]: Done.

**Comment by sergiyby** [comment]
I've been using a mixture of accounts to create CLs and this pull request, which seems to have confused the CLA check. I'll re-create the pull request from the same account as the author.

**Pull request closed without merging**